server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -297,6 +298,14 @@ public void persist(final Runnable commitRunnable)
 
     log.info("Submitting persist runnable for dataSource[%s]", schema.getDataSource());
 
+    if (persistExecutor.getQueue().size() >= config.getMaxPendingPersists()) {
```

Comment:
This will not be useful in the default configuration where maxPendingPersists = 0. It'll just print the warning every a persist is submitted, even if things are not actually being throttled. A better "possibly throttled" detection would need to check if a persist is running right before we submit a new one to the persistExecutor. That is racey but probably fine given that the check is only being used to emit a warning.